packages/openapi3-parser/lib/parser/oas/parseOpenAPIObject.js

```diff
@@ -99,6 +99,38 @@ function removeColumnLine(result) {
 const filterColumnLine = recurseSkippingAnnotations(removeColumnLine);
 const filterSourceMaps = recurseSkippingAnnotations(removeSourceMap);
 
+const deduplicateAnnotations = R.curry((namespace, parseResult) => {
+  // {str: Annotation[]} - indexes all warning annotations by message
+  const warnings = {}; // {str: Annotation[]}
+
+  const filterWarnings = R.ifElse(isWarningAnnotation,
+    (warning) => {
+      const annotations = warnings[warning.content];
+      if (annotations !== undefined) {
+        annotations.push(warning);
+        return false;
+      }
+
+      warnings[warning.content] = [warning];
+      return true;
+    },
+    R.T);
+
+  const result = new namespace.elements.ParseResult(R.filter(filterWarnings, parseResult));
+
+  // Update warning messages to include count
+  R.forEach(
+    (annotations) => {
+      const warning = annotations[0];
+      // eslint-disable-next-line no-param-reassign
+      warning.content = `${warning.content} (${annotations.length} occurances)`;
+    },
+    R.filter(annotations => annotations.length > 1, R.values(warnings))
+  );
+
+  return result;
+});
```

Comment:
This particular code doesn't work, but I can see how this could be approached with `reduce` with a few changes. I'll try and get it to work, possibly it won't be as fast as this will construct a new string warning message for each iteration (which can be thrown away next time). In the old code the looping over the final annotations at the end prevents that.

Reply:
If we are talking about potentially thousands, then indeed the performance hit might be noticeable.